src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -0,0 +1,290 @@
+        internal string AadIssuerV1 { get; set; }
+        internal string AadIssuerV2 { get; set; }
+        internal string AadAuthority { get; set; }
+
+        /// <summary>
+        /// Validate the issuer for single and multi-tenant applications of various audiences (Work and School accounts, or Work and School accounts +
+        /// Personal accounts) and the various clouds.
+        /// </summary>
+        /// <param name="issuer">Issuer to validate (will be tenanted).</param>
+        /// <param name="securityToken">Received security token.</param>
+        /// <param name="validationParameters">Token validation parameters.</param>
+        /// <remarks>The issuer is considered as valid if it has the same HTTP scheme and authority as the
+        /// authority from the configuration file, has a tenant ID, and optionally v2.0 (this web API
+        /// accepts both V1 and V2 tokens).</remarks>
+        /// <returns>The <c>issuer</c> if it's valid, or otherwise <c>SecurityTokenInvalidIssuerException</c> is thrown.</returns>
+        /// <exception cref="ArgumentNullException"> if <paramref name="securityToken"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"> if <paramref name="validationParameters"/> is null.</exception>
+        /// <exception cref="SecurityTokenInvalidIssuerException">if the issuer is invalid or if there is a network issue. </exception>
+        public string Validate(
+            string issuer,
+            SecurityToken securityToken,
+            TokenValidationParameters validationParameters)
+        {
+            _ = issuer ?? throw new ArgumentNullException(nameof(issuer));
+
+            _ = securityToken ?? throw new ArgumentNullException(nameof(securityToken));
+
+            _ = validationParameters ?? throw new ArgumentNullException(nameof(validationParameters));
+
+            string tenantId = GetTenantIdFromToken(securityToken);
+
+            if (string.IsNullOrWhiteSpace(tenantId))
+            {
+                throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidIssuerException(LogMessages.IDX40105));
+            }
+
+            if (validationParameters.ValidIssuers != null)
+            {
+                foreach (var validIssuerTemplate in validationParameters.ValidIssuers)
+                {
+                    if (IsValidIssuer(validIssuerTemplate, tenantId, issuer))
+                    {
+                        return issuer;
+                    }
+                }
+            }
+
+            if (validationParameters.ValidIssuer != null)
+            {
+                if (IsValidIssuer(validationParameters.ValidIssuer, tenantId, issuer))
+                {
+                    return issuer;
+                }
+            }
+
+            try
+            {
+                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                {
+                    if (AadIssuerV2 == null)
```

Comment:
In the v2 case can't TokenValidationParameters.ConfigurationManager and / or Configuration be used to get the issuer from: BaseConfiguration?

Reply:
yes. fixed. thanks for the pointing that out.